defaults/initrd.scripts

```diff
@@ -1074,7 +1074,7 @@ mount_devfs() {
 	fi
 
 	# Options copied from /etc/init.d/udev-mount, should probably be kept in sync
-	if ! fs_type_in_use devtmpfs
+	if ! grep -Eqse "^[^ ]+ /dev ${devfs} " /proc/mounts
```

Comment:
what you've proposed isn't quite the same.
- if `/foobar` was mounted with devtmpfs; then this block also wrongly did not fire, which is the bug you're trying to fix.
- previously, if devtmpfs was present, but /dev was mounted with regular tmpfs, this `if` block did NOT fire.

I think your grep test is expressed more clearly with:
awk -v devfs=$devfs  'BEGIN{rc=1}($2=="/dev" && $3==devfs){rc=0}END{exit rc}' /proc/mounts

If you agree and can rebase with that fix, i'll be happy to merge it, but I agree it shouldn't matter in normal boot cases.